test/e2e/quick_start_test.go

```diff
@@ -75,3 +77,48 @@ var _ = Describe("When following the Cluster API quick-start with Ignition", fun
 		}
 	})
 })
+
+var _ = Describe("When following the Cluster API quick-start check owner references are correctly reconciled and rereconciled if deleted", func() {
```

Comment:
What about integrating the new validation in the existing tests we already have?
(similar for the next test)

Reply:
I was thinking about letting them run in the CI / periodic for a while to understand the tests better and then following up on this - WDYT?

Reply:
Sounds very reasonable, please just track it in an issue.

I really try to keep the number of tests down, but given that quickstart runs on every PR it makes a lot of sense to check for stability in the periodics